### PR TITLE
Introduce service, copy and change tests, etc

### PR DESCRIPTION
**Pull request by MartinRijks**
[Task #105530](https://dev.azure.com/niposoftware/Nfield/_sprints/taskboard/Team%20Yellow/Nfield/Release%202022/2022%20P1S3?workitem=105530)

### CODE REVIEW

File: Tests/Services/NfieldCapiInterviewersServiceTests.cs [hunk 1]
```diff
@@ -0,0 +1,337 @@
...
+
+        [Fact]
+        public void TestRemoveAsync_ServerRemovedInterviewer_DoesNotThrow()
+        {
+            const string InterviewerId = "Interviewer X";
+            var interviewer = new Interviewer { InterviewerId = InterviewerId };
+            var mockedNfieldConnection = new Mock<INfieldConnectionClient>();
+            var mockedHttpClient = CreateHttpClientMock(mockedNfieldConnection);
+            mockedHttpClient
+                .Setup(client => client.DeleteAsync(new Uri(ServiceAddress, CapiInterviewersEndpoint + InterviewerId)))
```
Comment (ozaragoza) [inline Tests/Services/NfieldCapiInterviewersServiceTests.cs:77]:
```suggestion
                .Setup(client => client.DeleteAsync(new Uri(ServiceAddress, $"{CapiInterviewersEndpoint}{InterviewerId}")))
```

File: Tests/Services/NfieldCapiInterviewersServiceTests.cs [hunk 2]
```diff
@@ -0,0 +1,337 @@
...
+            var interviewerOut = new InterviewerChanged
+            {
+                InterviewerId = InterviewerId,
+                UserName = "XXX"
+            };
+
+            var mockedNfieldConnection = new Mock<INfieldConnectionClient>();
+            var mockedHttpClient = CreateHttpClientMock(mockedNfieldConnection);
+            mockedHttpClient
+                .Setup(client => client.PatchAsJsonAsync(new Uri(ServiceAddress, CapiInterviewersEndpoint + InterviewerId), It.IsAny<UpdateInterviewer>()))
```
Comment (ozaragoza) [inline Tests/Services/NfieldCapiInterviewersServiceTests.cs:118]:
```suggestion
                .Setup(client => client.PatchAsJsonAsync(new Uri(ServiceAddress, $"{CapiInterviewersEndpoint}{InterviewerId}"), It.IsAny<UpdateInterviewer>()))
```

File: Tests/Services/NfieldCapiInterviewersServiceTests.cs [hunk 3]
```diff
@@ -0,0 +1,337 @@
...
+        [Fact]
+        public void TestChangePasswordAsync_ServerChangesPassword_ReturnsInterviewer()
+        {
+            const string Password = "Password";
+            const string InterviewerId = "Interviewer X";
+            var interviewer = new Interviewer { InterviewerId = InterviewerId };
+            var mockedNfieldConnection = new Mock<INfieldConnectionClient>();
+            var mockedHttpClient = CreateHttpClientMock(mockedNfieldConnection);
+            mockedHttpClient
+                .Setup(client => client.PutAsJsonAsync(new Uri(ServiceAddress, CapiInterviewersEndpoint + InterviewerId), It.IsAny<object>()))
```
Comment (ozaragoza) [inline Tests/Services/NfieldCapiInterviewersServiceTests.cs:175]:
```suggestion
                .Setup(client => client.PutAsJsonAsync(new Uri(ServiceAddress, $"{CapiInterviewersEndpoint}{InterviewerId}"), It.IsAny<object>()))
```

File: Library/Models/InterviewerChanged.cs [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+        /// <summary>
+        /// Unique id of the interviewer
+        /// </summary>
+        [JsonProperty]
+        public string InterviewerId { get; internal set; }
+
+        /// <summary>
+        /// User name interviewer uses to sign in
+        /// </summary>
+        public string UserName { get; set; }
```
[the remark this answers is not in the document]
Comment (MartinRijks) [reply Library/Models/InterviewerChanged.cs:35]: Ah, probably yes :)

File: Tests/Services/NfieldCapiInterviewersServiceTests.cs (hunk 1 above)
Comment (MartinRijks) [reply Tests/Services/NfieldCapiInterviewersServiceTests.cs:77]: A clear sign of how I left old code, old code. :D Will change.

File: Library/Services/INfieldCapiInterviewersService.cs [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+
+        /// <summary>
+        /// Removes the interviewer.
+        /// </summary>
+        /// <param name="interviewer">The interviewer to remove.</param>
+        /// <exception cref="T:System.AggregateException"></exception>
+        /// The aggregate exception can contain:
+        /// <exception cref="Nfield.Exceptions.NfieldErrorException"></exception>
+        /// <exception cref="Nfield.Exceptions.NfieldHttpResponseException"></exception>       
+        Task RemoveAsync(Interviewer interviewer);
```
[the remark this answers is not in the document]
Comment (MartinRijks) [reply Library/Services/INfieldCapiInterviewersService.cs:46]: The original passed the interviewer object. I don't know the exact rationale behind it but I'm reluctant to change it as we would need to consider more changes where the Id is enough. This service was / is a bit messy in that aspect.

**Review by ozaragoza — APPROVED** [review]

**Pull request merged**